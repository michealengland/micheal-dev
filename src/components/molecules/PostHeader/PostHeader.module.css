.postHeader {
  & .inner {
    @apply text-primary-purple2 py-10 pb-4 px-4 max-w-3xl mx-auto font-semibold;
  }

  & .title {
    @apply text-4xl md:text-5xl;
  }

  & .content {
    @apply text-lg;
  }
}
