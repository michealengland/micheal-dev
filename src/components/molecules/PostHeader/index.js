import React from 'react'
import PropTypes from 'prop-types'
import Heading from '@components/atoms/Heading'

export default function PostHeader({content, title}) {
  return (
    <header>
      <Heading tag="h1">{title}</Heading>
      {content && <p>{content}</p>}
    </header>
  )
}

PostHeader.propTypes = {
  title: PropTypes.string,
  content: PropTypes.string,
}


