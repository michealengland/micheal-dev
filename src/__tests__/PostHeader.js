import React from 'react'
import renderer from 'react-test-renderer'
import PostHeader from '@components/molecules/PostHeader'

describe('PostHeader', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <PostHeader
          title="Post Header Title"
          content="Header content goes here..."
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
