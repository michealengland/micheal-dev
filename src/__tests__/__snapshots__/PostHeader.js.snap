// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostHeader renders correctly 1`] = `
<header
  className="postHeader"
>
  <div
    className="inner"
  >
    <h1
      className="title"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Post Header Title",
        }
      }
    />
    <p
      className="content"
    >
      Header content goes here...
    </p>
  </div>
</header>
`;
